### PR TITLE
added oneauth login with cookie session

### LINKED ISSUE #2: Add: User authentication via oneauth
Allow user authentication using oneauth. 

- `/login` and `/logout` endpoints using oneauth
- '/users/me` to get details of current user.

Can you use either JWT or cookies for session management.

### PR DESCRIPTION
**Pull request by architkshk**
fixes #2 @abhishek97  please review

### CODE REVIEW

**Comment by architkshk** [comment]
screenshot from 2018-05-23 17-55-25
Everything works fine for me.

File: util/auth.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      } else {
+          res.locals.isAuthenticated = false;
+      }
+      next();
+  },
+    checkToken(token,done){
+      rp({
+          uri : 'https://account.codingblocks.com/api/users/me',
+          headers : {
+              'Authorization' : `Bearer ${token}`
```
Comment (abhishek97) [inline util/auth.js]: So you expect the client app to send requests to Ide-backend using oneauth's access token?. 
No. Implement your own access tokens.

File: util/db.js [hunk 2]
```diff
@@ -1,14 +1,48 @@
...
+            host: secrets.HOST  || 'localhost',
+            dialect: 'postgres',
+            pool: {
+                max: 5,
+                min: 0,
+                idle: 10000
+            },
+        });
+
+const User = sequelize.define('user',{
```
Comment (abhishek97) [inline util/db.js]: Okay, we are fine with using sequelize here, like many of our other projects. However, please do use `sequelize-cli` or atleast some structure.

File: bin/www [hunk 3]
```diff
@@ -5,86 +5,89 @@
...
 var http = require('http');
 
 /**
  * Get port from environment and store in Express.
  */
+db.sync({force: false}).then(() => {
 
-var port = normalizePort(process.env.PORT || '3000');
-app.set('port', port);
+  var port = normalizePort(process.env.PORT || '3000');
```
Comment (abhishek97) [inline bin/www]: Too much irrelevant changes here.

Comment (architkshk) [reply bin/www]: Removed the changes

File: util/auth.js (hunk 1 above)
Comment (architkshk) [reply util/auth.js]: changed

File: routes/authRoutes.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+var db = require('../util/db');
+
+function getUserDetails(userId){
+    const details = db.User.findOne({id : userId});
+    return Promise.all([details]);
+};
+
+module.exports = app => {
+
+    app.get('/login', passport.authenticate('oauth2', { failureRedirect: '/failed' }) );
```
Comment (abhishek97) [inline routes/authRoutes.js]: > Also, I think you are mistaken here. We require Explicit Oauth2 flow here. We won't be having a callback handler on the server. The frontend will call the backend with the received grant_code and the backend just need to exchange that with access-token.
> 
> This is different from how auth works in boss-backend. You can take an example (frontend part only) from codingblocks online and hackerblocks.

**Pull request closed without merging**